### DISCUSSION TITLE
How to load google map api in a modal component livewire 3?

### DISCUSSION THREAD

**Discussion by awesome1128** [livewire/livewire#8092] [1. Help]
Hi everyone!
How are you?

I am trying to load google map api link in livewire modal component, but I assure that it's just asset link.
So, I added it in @assets directive , after that , I created callback function which will be calling from google map api.
So something like this.  This is a modal component blade file. I am using wire-element-modal package.

https://github.com/wire-elements/modal

```
<div>
    <div>
        <x-input-label for="geoAddress" :value="__('Search Address for Map')" />
        <input wire:model.blur="geoAddress" type="text" id="geoAddress" class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-primary-500 focus:border-primary-500 block w-full px-4 py-2 mt-2" required />
    </div>
    <div wire:ignore>
        <div id="address-map-container" class="mt-2 w-full h-[100px]">
            <div style="width: 100%; height: 100%" id="map"></div>
        </div>
    </div>
</div>

@assets
<script src="https://maps.googleapis.com/maps/api/js?key={My google map api key}&callback=initAutocomplete&libraries=places&v=weekly" defer></script>
@endassets

@script
<script>
function initAutocomplete() {
	// my code here.
}
</script>
@endscript
```

But when I open a modal, error is showing with "Uncaught (in promise) InvalidValueError: initAutocomplete is not a function".
I know this error why. It's because initAutomcomplete() function will be defined after fully loaded links in @assets.
Livewire 3 document mentions that @script directive will be loaded after fully loaded links in @assets.

https://livewire.laravel.com/docs/javascript#loading-assets

So, I tried to put google map api link like this.  But I know this will not work.
```
@script
<script>
function initAutocomplete() {
	// my code here.
}
</script>
<script src="https://maps.googleapis.com/maps/api/js?key={My google map api key}&callback=initAutocomplete&libraries=places&v=weekly" defer></script>
@endscript
```

Then, I assume , there is no way to load google map api link.
Please help me how to load api link or let me know if there is any google map api package that I can use with livewire 3.
Thank you so much!

**Comment by danie-ramdhani** [comment]
you got error `Uncaught (in promise) InvalidValueError: initAutocomplete is not a function` because that function is not available for global/window. 

try:

```blade

<div x-data="{ 
   init(){
      // your code here
   }
}">

</div>

@assets
<script src="https://maps.googleapis.com/maps/api/js?key={My google map api key}&callback=initAutocomplete&libraries=places&v=weekly" defer></script>
@endassets
```

> livewire is bundled with alpine js, learn here https://alpinejs.dev/directives/data
